### PR TITLE
Use GHA to add code static analysis with quality gating over SonarQube

### PR DESCRIPTION
**Pull request by konstan**
- [x] add Clojure static code analysis plugins
- [x] add GHA with sonar-clojure and publication to SixSq SonarQube
- [x] add JUnit in XML reporter and publication of the test results on the PR
- [x] rename TRAVIS_ to CI_

For explanations see https://github.com/SixSq/tasklist/wiki/Code-analysis#implementation-for-clojure

Related to https://github.com/SixSq/tasklist/issues/1982

### CODE REVIEW

File: code/project.clj [hunk 1]
```diff
@@ -10,7 +10,13 @@
             :url "http://www.apache.org/licenses/LICENSE-2.0.txt"
             :distribution :repo}
 
-  :plugins [[lein-parent "0.3.5"]]
+  :plugins [[lein-parent "0.3.5"]
```
Comment (0xbase12) [inline code/project.clj:13]: I think it's better to have the plugins in parent-project. Such way it will be updated when we update parent project

**Review by 0xbase12 — CHANGES_REQUESTED** [review]

File: code/project.clj (hunk 1 above)
Comment (konstan) [reply code/project.clj:13]: Yes, agreed! This is one of the topics I wanted to discuss.

File: code/project.clj [hunk 2]
```diff
@@ -36,6 +42,7 @@
    [org.slf4j/slf4j-api]
    [org.slf4j/slf4j-log4j12]
    [ring/ring-core]                                         ;; added to force version
+   [clj-kondo "RELEASE"]
```
[the remark this answers is not in the document]
Comment (konstan) [reply code/project.clj:45]: Agreed.

File: code/project.clj (hunk 1 above)
Comment (konstan) [reply code/project.clj]: The plugins were moved to parent.

File: code/project.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (konstan) [reply code/project.clj]: `:dev` profile is used now.

**Comment by konstan** [comment]
Update to run only on merge.

**Pull request merged**